graphium/fingerprinting/data.py

```diff
@@ -0,0 +1,216 @@
+from typing import Any, List, Dict, Literal, Union
+
+import os
+
+import torch
+
+import pandas as pd
+import numpy as np
+
+from pytorch_lightning import LightningDataModule
+
+from torch.utils.data import Dataset, DataLoader
+
+from graphium.data.datamodule import MultitaskFromSmilesDataModule, ADMETBenchmarkDataModule
+from graphium.trainer.predictor import PredictorModule
+from graphium.fingerprinting.fingerprinter import Fingerprinter
+
+
+class FingerprintDataset(Dataset):
+    def __init__(
+            self,
+            smiles: List[str],
```

Comment:
Shouldn't the Smiles be the last item, and be optional? Since they're not returned, unless it's for debugging or testing purposes?
And if `smiles is not None`, than also return the smiles in the `__getitem__`? Otherwise there's no point in having the smiles

Reply:
Agree and made it optional [here](https://github.com/datamol-io/graphium/pull/529/commits/d9aa407d774934e170e8d8af7028155b676b5981).